### DISCUSSION TITLE
How to debug tokio/tasks with only a core dump ?

### DISCUSSION THREAD

**Discussion by erebe** [tokio-rs/tokio#5378] [Q&A]
Hello,

I am looking for some insight/guidance regarding how to debug tokio program with only access to a coredump.

I have a program that mysteriously stop doing anything useful, after some investigation, it seems that the task responsible for running the useful stuff is never becoming ready/never scheduled.
I have a coredump of this application, and I would like to know where should I look inside tokio to find my task and poke at its internal in order to try to understand its current state.
Is that even possible ?
Any information regarding how to debug current state with a debugger will be welcome :pray: 

P.s: Thank you for tokio it is an awesome framework/library for concurrent programming.

**Comment by Noah-Kennedy** [comment]
Assuming that you compiled your program with all of the debug symbols, this is possible although very, very hard.
Locating the task is a pain to do, so I usually look for something which I know would have a waker to the task and use the waker to find the task. Once you have that, actually understanding what the future is doing and where it is in its lifecycle is also a pain, but it can be done.

**Comment by erebe** [comment]
Hello,

Thank you for your answer, I was fearing the same.
If by any chances you can write an article/record a video of yourself doing/explaining it, please let me know.
I can even sponsor this work if you have a PayPal, let say $100 for a video :P ?
